From a7986903a9158b8a0e5f7c5d5e979bc3929bde1e Mon Sep 17 00:00:00 2001
From: xtex <xtex@astrafall.org>
Date: Sun, 8 Mar 2026 10:30:02 +0800
Subject: [PATCH 131/133] AOSC OS: add custom dev envs
X-Developer-Signature: v=1; a=openpgp-sha256; l=2552; i=xtex@astrafall.org;
 h=from:subject; bh=R6RIunsJI/hlCEVXOBQGMJU5hoELOg/hA0sHwRsQs+k=;
 b=owGbwMvMwCW2U4Ij7wZL9ETG02pJDJn78gQyVFmcVEy/vvh5cvMORtXMWy1S8wxdyzjuGDBPW
 hb2oXh5x0QWBjEuBksxRZYiwwZvVp10ftFl5bIwc1iZQIZIizQwMDAwsDDw5SbmlRrpGOmZahvq
 GRrqGOkYMXBxCsBUr+VjZHhiXjDDx3STlMKOW98NbvnaO80XC+fVc2Zertio8rlA2Zrhf+xWl6/
 nPBwqbDlsvW7V1a75MIdzzkdB4+P5Ht9nv2qRZAMA
X-Developer-Key: i=xtex@astrafall.org; a=openpgp;
 fpr=7231804B052C670F15A6771DB918086ED8045B91

Change-Id: I3c22930523ec931e41d530e7ad62dd0ffd1bbf54
---
 src/dev.zig  | 38 ++++++++++++++++++++++++++++++++++++++
 src/main.zig |  2 ++
 2 files changed, 40 insertions(+)

diff --git a/src/dev.zig b/src/dev.zig
index ac30cedecd83..63d1ae545b72 100644
--- a/src/dev.zig
+++ b/src/dev.zig
@@ -52,6 +52,9 @@ pub const Env = enum {
     /// - `zig build-* -fincremental -fno-llvm -fno-lld -target x86_64-linux --listen=-`
     @"x86_64-linux",
 
+    @"aosc-bootstrap",
+    @"aosc-detect-cpu",
+
     pub inline fn supports(comptime dev_env: Env, comptime feature: Feature) bool {
         return switch (dev_env) {
             .full => true,
@@ -213,6 +216,41 @@ pub const Env = enum {
                 => true,
                 else => Env.sema.supports(feature),
             },
+            .@"aosc-bootstrap" => switch (feature) {
+                .build_exe_command,
+                .build_lib_command,
+                .build_obj_command,
+                .make_executable,
+                .make_writable,
+                .build_command,
+                .stdio_listen,
+                .incremental,
+                .ast_gen,
+                .sema,
+                .legalize,
+                .llvm_backend,
+                .lld_linker,
+                .detect_cpu_command,
+                => true,
+                else => Env.sema.supports(feature),
+            },
+            .@"aosc-detect-cpu" => switch (feature) {
+                .build_exe_command,
+                .build_lib_command,
+                .build_obj_command,
+                .make_executable,
+                .make_writable,
+                .build_command,
+                .stdio_listen,
+                .incremental,
+                .ast_gen,
+                .sema,
+                .legalize,
+                .llvm_backend,
+                .lld_linker,
+                => true,
+                else => Env.sema.supports(feature),
+            },
         };
     }
 
diff --git a/src/main.zig b/src/main.zig
index fd989cdc569d..887c9777ead7 100644
--- a/src/main.zig
+++ b/src/main.zig
@@ -5200,6 +5200,8 @@ fn cmdBuild(gpa: Allocator, arena: Allocator, args: []const []const u8) !void {
                 .have_zcu = true,
                 .emit_bin = true,
                 .is_test = false,
+
+                .use_llvm = if (dev.env == .@"aosc-bootstrap") true else null,
             });
 
             const root_mod = try Package.Module.create(arena, .{
-- 
2.52.0

